File: 08-DesignPatterns-Java/Starter/State/src/Partner.java
public class Partner {
    private PartnerState _state = PartnerState.NEUTRAL;

    public void talk() {
        switch (_state) {
            case NEUTRAL:
                print("Hello!");
                break;

            case HAPPY:
                print("I am so happy!");
                break;

            case ANGRY:
                print("Go away, I am angry!");
                break;

            default:
                throw new IllegalArgumentException();
        }
    }

    public void kiss() {
        switch (_state) {
            case NEUTRAL:
                print("Thanks! :-)");
                setState(PartnerState.HAPPY);
                break;

            case HAPPY:
                print("Thanks! <3 <3");
                break;

            case ANGRY:
                print("Okay, I forgive you.");
                setState(PartnerState.NEUTRAL);
                break;

            default:
                throw new IllegalArgumentException();
        }
    }

    public void insult() {
        switch (_state) {
            case NEUTRAL:
                print("Are you f***ing kidding me?!");
                setState(PartnerState.ANGRY);
                break;

            case HAPPY:
                print("Why would you say something like that? :-(");
                setState(PartnerState.ANGRY);
                break;

            case ANGRY:
                print("You shall burn in hell!");
                break;

            default:
                throw new IllegalArgumentException();
        }
    }

    private void print(String message) {
        System.out.println(message);
    }

    private void setState(PartnerState newState) {
        _state = newState;
    }
}

File: 08-DesignPatterns-CSharp/08-1-2-State/Partner.cs
namespace State
{
    using System;

    public class Partner
    {
        private PartnerState _state = PartnerState.Neutral;

        public void Talk()
        {
            switch (_state)
            {
                case PartnerState.Neutral:
                    Print("Hello!");
                    break;

                case PartnerState.Happy:
                    Print("I am so happy!");
                    break;

                case PartnerState.Angry:
                    Print("Go away, I am angry!");
                    break;

                default:
                    throw new ArgumentOutOfRangeException();
            }
        }

        public void Kiss()
        {
            switch (_state)
            {
                case PartnerState.Neutral:
                    Print("Thanks! :-)");
                    SetState(PartnerState.Happy);
                    break;

                case PartnerState.Happy:
                    Print("Thanks! <3 <3");
                    break;

                case PartnerState.Angry:
                    Print("Okay, I forgive you.");
                    SetState(PartnerState.Neutral);
                    break;

                default:
                    throw new ArgumentOutOfRangeException();
            }
        }

        public void Insult()
        {
            switch (_state)
            {
                case PartnerState.Neutral:
                    Print("Are you f***ing kidding me?!");
                    SetState(PartnerState.Angry);
                    break;

                case PartnerState.Happy:
                    Print("Why would you say something like that? :-(");
                    SetState(PartnerState.Angry);
                    break;

                case PartnerState.Angry:
                    Print("You shall burn in hell!");
                    break;

                default:
                    throw new ArgumentOutOfRangeException();
            }
        }

        private void Print(string message) => Console.WriteLine(message);

        private void SetState(PartnerState newState) => _state = newState;
    }
}